src/agent/application.cpp

```diff
@@ -60,45 +60,50 @@ Application::Application(const std::string               &aInterfaceName,
 #else
     , mBackboneInterfaceName(aBackboneInterfaceNames.empty() ? "" : aBackboneInterfaceNames.front())
 #endif
-    , mHost(mInterfaceName.c_str(), aRadioUrls, mBackboneInterfaceName, /* aDryRun */ false, aEnableAutoAttach)
 #if OTBR_ENABLE_MDNS
     , mPublisher(Mdns::Publisher::Create([this](Mdns::Publisher::State aState) { this->HandleMdnsState(aState); }))
 #endif
+#if OTBR_ENABLE_VENDOR_SERVER
+    , mVendorServer(vendor::VendorServer::newInstance(*this))
+#endif
+{
+    mHost = std::unique_ptr<Ncp::RcpHost>(new Ncp::RcpHost(mInterfaceName.c_str(), aRadioUrls, mBackboneInterfaceName,
```

Comment:
I think you can use https://github.com/openthread/ot-br-posix/blob/5a7972eb22434f3eea2d1651bfd6a3e60f0821ba/src/common/code_utils.hpp#L167